src/CSnakes.SourceGeneration/PythonStaticGenerator.cs

```diff
@@ -107,6 +117,18 @@ private class {{pascalFileName}}Internal : I{{pascalFileName}}
                         }
                     }
 
+                    /// <summary>
+                    /// Reload the module, useful for development if the Python code has changed.
+                    /// </summary>
+                    public void ReloadModule() {
+                        logger.LogDebug("Reloading module {ModuleName}", "{{fileName}}");
+                        using (GIL.Acquire())
+                        {
+                            module = Import.ReloadModule(module);
+                            {{string.Join(Environment.NewLine, functionNames.Select(f => $"this.{f.Field} = module.GetAttr(\"{f.Attr}\");"))}}
```

Comment:
This could go in a separate method that's shared with the same code being generated in the constructor so it's JIT'd once.

More importantly though, I see a few concerns here:

- Won't this leak the previous function objects since the fields just get overwritten with potentially new objects?
- Is it exception-safe? What if the initialisation of the fields fails halfway? Won't it leave `this` in some strange state?
- What happens if a function was removed, updated (the signature rather than the code) or added?

I think the supported scenarios should be carefully documented to keep expectations in check.

Reply:
> Won't this leak the previous function objects since the fields just get overwritten with potentially new objects?

The original object will get GC'ed and collected

> Is it exception-safe? What if the initialisation of the fields fails halfway? Won't it leave this in some strange state?

The exceptions are tied to the frame, which is associated to the code object. IF we still hold a reference to the old code object, we would need to close it.

> What happens if a function was removed, updated (the signature rather than the code) or added?

Hot Reload tells you that you need to restart (I tried this) and there are tests for that as well now